app/policies/comment_policy.rb

```diff
@@ -0,0 +1,24 @@
+class CommentPolicy < ApplicationPolicy
+  DELETE_COMMENT_ROLES = %w(superadmin policy_and_abuse communications support).freeze
+
+  def self.can_delete_comment?(user)
+    self.new(user, nil).can_delete_comment?
+  end
+    
+  def can_delete_comment?
+    user_has_roles?(DELETE_COMMENT_ROLES)
+  end
+
+  def can_mark_comment_spam?
+    (can_delete_comment? || is_author_of?(record.ultimate_parent))
+  end
+
+  alias_method :destroy?, :can_delete_comment?
+  alias_method :reject?, :can_mark_comment_spam?
+
+  private
+
+  def is_author_of?(work)
+    user.is_a?(User) && user.is_author_of?(work)
```

Comment:
Should we have this check, since all the views appear to use the policy with `logged_in_as_admin? &&`?

Reply:
This policy is used in the comments controller, in the reject method. This controller method is used in the admin section, but also on the regular posts, where (at least I'm under the impression) admins and authors can mark spam on the comments for their posts. So this policy check that an admin is trying first, and then if the author is trying to use it. Does that sound like the intended functionality?

Reply:
Yup, you're right: admins and work creators can both mark comments as spam. (That's true with deleting comments, too. However, the author of the _comment_ can also delete it, assuming they're a user and not a guest.)

Reply:
Having this check here is kinda confusing to me, because we're using Pundit for both admin roles and user relationship to item (in this case, whether or not they have ownership of works).

Is it possible to involve Pundit policies only when logged in as an admin? For example, can we work with `pundit_user` returning `current_admin` instead of `current_admin || current_user` (returning nil if not logged in as an admin)?

Reply:
red makes a good point: we don't intend to use Pundit for regular users, so `pundit_user` should only ever be an admin, and any checks related to regular users should stay where they previously were.